Exoft.Gamification.Api.Data/Repositories/AchievementRepository.cs

```diff
@@ -14,16 +15,43 @@ public AchievementRepository(UsersDbContext context) : base(context)
         {
         }
 
-        public async Task<ICollection<Achievement>> GetPagedAchievementAsync(PageInfo pageInfo)
+        public async Task<ReturnPagingInfo<Achievement>> GetPagedAchievementByUserAsync(PagingInfo pagingInfo, Guid UserId)
         {
-            var list = await IncludeAll().Skip((pageInfo.PageNumber - 1) * pageInfo.PageSize).Take(pageInfo.PageSize).ToListAsync();
+            var list = await Context.UserAchievements
+                .Where(o => o.User.Id == UserId)
+                .Select(i => i.Achievement)
+                .ToListAsync();
 
-            return list;
+            var items = list
+                .Skip((pagingInfo.CurrentPage - 1) * pagingInfo.PageSize)
+                .Take(pagingInfo.PageSize)
+                .ToList();
+
+            var result = new ReturnPagingInfo<Achievement>()
+            {
+                CurrentPage = pagingInfo.CurrentPage,
+                PageSize = items.Count,
+                TotalItems = list.Count,
+                TotalPages = (int)Math.Ceiling((double)list.Count / pagingInfo.PageSize),
+                Data = items
+            };
+
+            return result;
         }
 
         protected override IQueryable<Achievement> IncludeAll()
         {
-            return DbSet.Include(i => i.Icon);
+            return DbSet
+                .Include(i => i.Icon)
+                .OrderByDescending(i => i.XP);
```

Comment:
I'd not recommend you to put order here.... Somewhere it would ne redundant command on SQL serer.... also method should do Include... not order...